### PR TITLE
use X, Y, Z for Lon, Lat, Vert

### LINKED ISSUE #140: Write GDALarray with X-Y coordiates
Currently I get:
```
julia> write("/tmp/t.tif", GDALarray, ar)
ERROR: Array must have Lat and Lon dims
...
```
As my `ar` is in a projected coordinate system it has X-Y dims:
```
julia> dims(ar)
(X (type X): Int32[-1800000, -1799500, …, -960500, -960000] (Sampled: Ordered Regular Intervals), Y (type Y): Int32[-880000, -879500, …, 139500, 140000] (Sampled: Ordered Regular Intervals))
```

Is this a missing feature or am I missing something?

### LINKED ISSUE #145: Just use X, Y and Z and drop Lat/Lon and all Geo dims
I've been thinking about this since starting the package. But #140 shows it's probably the way to go, and `Lat` and `Lon` really don't make sense.

Plotting would then assume that all `XDim/YDim/ZDim` are spatial and should be a heatmap for all `AbstractGeoArray`, which is probably reasonable.

### PR DESCRIPTION
**Pull request by rafaqz**
@mauro3 mostly `sed` took care of this ;)

I'm pretty keen on making this change now, but I'll leave this here a while for comments.

Closes #140 
Closes #145

### CODE REVIEW

**Comment by rafaqz** [comment]
@jamesmaino you probably have an opinion on this

**Comment by jamesmaino** [comment]
Yep, I guess it needs this to handle arbitrary CRS rasters. 

Hopefully it doesn't break too many of our scripts!

**Comment by rafaqz** [comment]
I'll do `const Lat=Y` etc to make it a bit less painful.

File: src/sources/ncdatasets.jl [hunk 1]
```diff
@@ -48,20 +48,21 @@ const NCD_FILL_TYPES = (Int8,UInt8,Int16,UInt16,Int32,UInt32,Int64,UInt64,Float3
...
+    "long" => X,
+    "longitude" => X,
+    "time" => Ti,
+    "lev" => Z,
+    "mlev" => Z,
+    "level" => Z,
+    "vertical" => Z,
+    "x" => X,
+    "y" => Y,
+    "z" => Z,
```
Comment (vlandau) [inline src/sources/ncdatasets.jl:64]: Will this handle capitalized dim names? Maybe you're already doing a "lowercase" somewhere, or maybe dims are already usually lowercase? @rafaqz

**Review by vlandau — APPROVED** [review]
Looks like a pretty straight forward change to me. I think it makes a lot of sense to switch over from Lat and Lon -- Y and X (and Z) is much more general (`A[Y(Between(y1, y2)), X(Between(x1, x2))]` seems nicer and much more natural, too).

Only clarification is that this doesn't change anything related to `Band` or `Ti`, since those are not `<: ZDim` right?

Also, is there any need for deprecation warnings? Just to anticipate one day removing `CONST Lat = Y`, etc., though I see how this could get tedious to implement. Not really a big deal to keep those CONSTs in there though...

I noticed a tiny typo in a docstring so I did a PR to this branch to fix it. Other than that, looks good to go!

File: src/dimensions.jl [hunk 2]
```diff
@@ -1,81 +1,3 @@
...
-Abstract supertype for GeoSpatial Y dimensions. 
-"""
-abstract type GeoYDim{T,Mo,Me} <: YDim{T,Mo,Me} end
-
-"""
-    GeoZDim <: Dimension.ZDim
-
-Abstract supertype for GeoSpatial Z dimensions. 
-"""
-abstract type GeoZDim{T,Mo,Me} <: ZDim{T,Mo,Me} end
```
[the remark this answers is not in the document]
Comment (rafaqz) [reply src/dimensions.jl:20]: Not a dumb question at all! the distinction I made by writing them was probably dumb though... Now it just uses `XDim/YDim/ZDim` from DD, most importantly in plot reorientation. We do still need some abstraction level (like `X <: XDim`) so you can define other dimensions to act like `X/Y/Z`. E.g. they might have a different/unknown name in netcdf and then they wont plot the right way unless we define `@dim WeirdDim XDim`.

**Comment by rafaqz** [comment]
Thanks for the review, I appreciate the feedback. Yes `Band` is just `<: Dim` as defined here, and `Ti` is `<: IndependentDim` so it does still go on the X axis when possible - but it's defined in DD anyway.

And I don't really know how to do deprecations for things like this! how to get a warning when you use a constant? I guess we could make it a function instead? I rarely use deprecations because I don't get how to do it cleanly, but would like to learn if you have ideas.

**Comment by vlandau** [comment]
Yeah I suppose it would have to be a function -- in which case [`@deprecate`](https://docs.julialang.org/en/v1/base/base/#Base.@deprecate) would be the way to go; however, as of Julia 1.5, users won't see the warnings (outside of package tests) unless they run Julia with `--depwarn=yes`. Maybe not worth the trouble?

**Comment by mauro3** [comment]
I would favor to remove `Lat` `Lon` at some stage as it is nicer if there is only one way to do things to have code consistent looking.  But no big deal either way.

**Comment by rafaqz** [comment]
Yep they will get removed fairly soon. Its just that as @jamesmaino mentions above, at cesar we have a lot of scripts in progress that shouldnt just break from this change, and I imagine others do as well.

I will use functions and `@depreciate`

**Comment by rafaqz** [comment]
Turns out it's hard to depreciate `Lat/Lon` because a lot of the time they are just used as types, without being constructed... so there is no function to give a depreciation warning.

**Comment by vlandau** [comment]
Okay I thought it might be tricky -- as long as the docs now use X/Y/Z I think it's fine to plan to get rid of Lat/Lon/Vert for the first major release (or even next minor since 0.4.0 -> 0.5.0 can be considered breaking)

**Comment by mauro3** [comment]
Excellent, thanks!

**Pull request merged**